lib/query.js

```diff
@@ -165,6 +165,32 @@ function Query(conditions, options, model, collection) {
   }
 }
 
+// Helper function to check if an object is empty or contains only empty objects/arrays
+function isEmptyFilter(obj) {
+  if (obj == null) return true;
+  if (typeof obj !== 'object' || obj === null) return false;
```

Comment:
What is this check about? `obj === null` will never be true here. And I think that if `typeof obj !== 'object'` we would return true here.